app/main.py

```diff
@@ -2,0 +1,38 @@
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __str__(self) -> str:
+        return (f"Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}"
+                )
+
+    def __repr__(self) -> str:
+        return f"{{{self}}}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden :
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                Animal.alive.remove(herbivore)
```

Comment:
Before removing `herbivore` from `Animal.alive`, ensure that `herbivore` is actually in the list to avoid potential `ValueError` exceptions. Consider using a check like `if herbivore in Animal.alive:` before attempting to remove it.